### ISSUE TITLE
IconMap annotation is ignored when using short class definition for a connector

### ISSUE THREAD

**Issue by gwr69** [OpenModelica/OpenModelica#12097]
### Description

Using `IconMap` annotation to prevent a connector to inherit graphical annotations for its icon layer does not work out when using short class definition.

### Steps to Reproduce

Consider the following minimal example:

```modelica
package DemoPackage

    type Reals = Real annotation(
    Icon(graphics = {Text(origin = {2, 2}, textColor = {255, 0, 0}, extent = {{-32, 34}, {32, -34}}, textString = "R", fontName = "Arial")}));

  connector RealOutput = output Reals annotation(IconMap(primitivesVisible = false));

end DemoPackage;
```

and also this variant:

```modelica
package DemoPackage

    type Reals = Real annotation(
    Icon(graphics = {Text(origin = {2, 2}, textColor = {255, 0, 0}, extent = {{-32, 34}, {32, -34}}, textString = "R", fontName = "Arial")}));

  connector RealOutput = output Reals annotation(IconMap(extent = {{1,1},{1,1}}));

end DemoPackage;
```

### Expected Behavior

According to [Section 18.6.3](https://specification.modelica.org/maint/3.6/annotations.html#extends-clause) of the Modelica Language Specification I am expecting to see an empty icon layer for `RealOutput` in both examples, yet the connector will have inherited the red letter `"R"` in both examples.

### Version and OS

 - OpenModelica Version: `1.23.0-dev-368-g5596fcba0f (64-bit)`
 - OS: `Windows 10, 64 bit`

### Additional Context

Similar issue noted for class `Interfaces.Basics.OutputTypeChoice.RealOutput` of https://github.com/bslMS/BusinessSimulation/tree/release when trying to make icon layer annotation for base type "disappear".

**Comment by adeas31** [comment]
Turns out changing this in the API was not easy. PR #12099 should fix this.

**Comment by gwr69** [comment]
I just modified the three OutputTypeChoice-related partial classes `OutputTypeChoice`, `OutputTypeChoice_Rate`, and `OutputTypeChoice_Time`. For examples:

https://github.com/bslMS/BusinessSimulation/blob/release/BusinessSimulation/Interfaces/Basics/OutputTypeChoice.mo

where I added `IconMap(primitivesVisible = false)` for the `type` component:

```
encapsulated partial class OutputTypeChoice "Partial model to set output type"
  import BusinessSimulation.Units.*;
  replaceable type OutputType = Unspecified constrainedby Unspecified "Type choice" annotation(IconMap(primitivesVisible = false),choicesAllMatching = true);
  connector RealOutput = output OutputType "Output signal" annotation(...)
...
```

The `IconMap` annotation will not have any effect, i.e., "peeking" under the triangle in the icon layer for `connector RealOutput` shows the letter `"R"` from the `type` component. What am I missing?

I am now using todays nightly build.
